### PR TITLE
Fix Rust lint CI

### PR DESCRIPTION
**Pull request by erikjohnston**
We weren't correctly reporting warnings as errors.

### CODE REVIEW

File: changelog.d/14602.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix Rust ling CI.
```
Comment (DMRobertson) [inline changelog.d/14602.misc:1]:
```suggestion
Fix Rust lint CI.
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -109,7 +109,7 @@ jobs:
             components: clippy
       - uses: Swatinem/rust-cache@v2
 
-      - run: cargo clippy
+      - run: cargo clippy --all-targets -- -D warnings
```
Comment (erikjohnston) [inline .github/workflows/tests.yml:112]: `--all-targets` makes sure we check the `test` and `benches` stuff too

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -109,7 +109,29 @@ jobs:
             components: clippy
       - uses: Swatinem/rust-cache@v2
 
-      - run: cargo clippy
+      - run: cargo clippy -- -D warnings
```
Comment (DMRobertson) [inline .github/workflows/tests.yml:112]: guessing this is analogous to `--Werror`?

Are there any other flags we should turn on?

**Review by DMRobertson — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 3 above)
Comment (erikjohnston) [reply .github/workflows/tests.yml:112]: Yeah, its just "deny all warnings". I think the defaults are relatively sensible TBH.

**Pull request merged**